poetry/core/masonry/builders/wheel.py

```diff
@@ -91,44 +96,59 @@ def build(self):
 
     def _build(self, wheel):
         if self._package.build_script:
-            with SdistBuilder(poetry=self._poetry).setup_py() as setup:
-                # We need to place ourselves in the temporary
-                # directory in order to build the package
+            if not self._poetry.package.build_config.get("generate-setup-file", True):
```

Comment:
We should probably make this a property of the package (`package.build_generate_setup_file`) or have something like `package.build.generate_setup_file`; we can handle this later if required but otherwise we will end up with multiple places to update default value etc.